### ISSUE TITLE
Cannot enter dates when adding a new client

### ISSUE THREAD

**Issue by lamontfr** [savoirfairelinux/sous-chef#726]
### Expected Behaviour

- When adding a new client, must choose or enter birthdate and referral date.

### Actual Behaviour
- In chrome and firefox, cannot select birth date in calendar, and when entering date in field pressing tab clears the date; but pressing enter accepts the date.
- In chrome and firefox, cannot select referral date in calendar, and when entering date in field pressing tab clears the date.
- In IE 11,  cannot select birth date and referral date in calendar, and when entering date in field pressing tab clears the date.

### Notes
- This was working OK on 2017-03-31 when PR 711 was pulled (commit fd8e360)
- I tested with dev after commit  04044df when PR 722 was pulled on 2017-04-09 and it is broken.

**Comment by lamontfr** [comment]
@lingxiaoyang I just tried again, here are the two cases.

Case 1
- git checkout dev (PR 719 commit 4035144)
- restart docker webserver
- run /code/tools/gulp#  gulp
- clear browser cache
- sous chef new client : cannot enter birthdate : BUG

Case 2
- git checkout fix696 (my branch based on PR 711 commit fd8e360)
- restart docker webserver
- run /code/tools/gulp#  gulp
- clear browser cache
- sous chef new client : can enter dates properly : OK

Am I missing something ?

**Comment by lingxiaoyang** [comment]
@lamontfr  I tried again with 4035144 and it seemed working fine with chrome and firefox. I could select a date from the calendar and also directly put the date in the text box (yyyy-mm-dd format). 

I'll let other people test it.

**Comment by lamontfr** [comment]
@lingxiaoyang When I use the calendar, get a "reference error" in the js console : dateformat is not defined. However, gulpfile.js contains : 
```js
 js: {
    scripts: {
      vendor: [
        'node_modules/jquery-tablesort/jquery-tablesort.js',
        'node_modules/jquery.formset/src/jquery.formset.js',
        'node_modules/semantic-ui-calendar/dist/calendar.js',
        'node_modules/dateformat/lib/dateformat.js',
      ],
```
and I ran gulp after the checkout. Here is the screenshot
![screenshot at 2017-04-18 08-31-39](https://cloud.githubusercontent.com/assets/8512608/25130976/249b3118-2412-11e7-9b7d-ae1f3a3b42c4.png)

Any idea ?

**Comment by lingxiaoyang** [comment]
@lamontfr  That's certainly because your browser didn't load dateformat package for some reason. During `gulp`, this package should be compiled into `http://localhost:8000/static/js/sous-chef.min.js`.

@peakwinter Do you have an idea of this?

**Comment by peakwinter** [comment]
@lingxiaoyang @lamontfr If you haven't updated your NPM packages since that last commit was added, that would certainly cause this kind of issue.

`docker-compose exec web sh -c "cd tools/gulp && npm install"`

**Comment by lamontfr** [comment]
@lingxiaoyang @peakwinter Thanks for the hints, I will try it tonight.

**Comment by lingxiaoyang** [comment]
@lamontfr  I just made a hotfix. There was something missing in the PR that added `dateformat`. Please try with the latest dev branch.

**Comment by lamontfr** [comment]
@lingxiaoyang @peakwinter Thanks to both of you. PR #727 fixes the problem. I just had to update the NPM packages, ran gulp and cleared the cache. I am closing the issue.

**Comment by lamontfr** [comment]
Fixed by PR #727.
